src/test/java/duke/ParserTest.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
+    @Test
+    public void parser_commandTypeBye_expectedBehaviour() {
+        Parser parser = new Parser();
+        try {
+            Command byeCommand = parser.parse("bye");
+            assert(byeCommand instanceof ByeCommand);
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    public void parser_commandTypeList_expectedBehaviour() {
+        Parser parser = new Parser();
+        try {
+            Command listCommand = parser.parse("list");
+            assert(listCommand instanceof ListCommand);
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    public void parser_commandTypeDone_expectedBehaviour() {
+        Parser parser = new Parser();
+        try {
+            Command doneCommand = parser.parse("done 2");
+            assert(doneCommand instanceof DoneCommand);
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    public void parser_commandTypeDelete_expectedBehaviour() {
+        Parser parser = new Parser();
+        try {
+            Command deleteCommand = parser.parse("delete 2");
+            assert(deleteCommand instanceof DeleteCommand);
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    public void parser_commandTypeDelete_exceptionThrown() {
+        Parser parser = new Parser();
+        assertThrows(DukeException.class, () -> parser.parse("delete"));
+    }
```

Comment:
I like the test coverage of `DukeException`. Perhaps can further test if the `DukeException` display the respective error message.